### DISCUSSION TITLE
Will blackmagic firmware be considered for zephyr development in the future?

### DISCUSSION THREAD

**Discussion by Cherish-Gww** [blackmagic-debug/blackmagic#1251] [Ideas]
https://github.com/zephyrproject-rtos/zephyr

**Comment by dragonmux** [comment]
I'm not sure what you mean, however:

* We have support for debugging targets that are running an RTOS through GDB sidecar scripts - though it's possible those have bitrot and/or don't support Zephyr, and we recognise they need properly pulling together under the project banner so they're easier to find.
* We are under certain restrictions (particularly Flash space) that mean using an RTOS to express the firmware would be difficult (alongside the complexity and structure of the firmware not lending itself to an RTOS delivering any benefits).

Does this answer your question?

An example re the sidecar scripts are the [chibios](https://github.com/gsmcmullin/gdb_chibios) RTOS ones.
